### PR TITLE
Feature user display

### PR DESCRIPTION
**Pull request by witwitwut**
I would like to head some feedback, on the code and some of the redundancies I have snagged on.

### CODE REVIEW

File: src/store/modules/user.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ export const state = {
 export const mutations = {
   SET_CURRENT_USER(state, currentUser) {
     // console.log(state, currentUser)
-    state.currentUser = currentUser
+    state.currentUser =currentUser
```
Comment (Doommer43) [inline src/store/modules/user.js]: Add spacing for consistency.

File: src/store/modules/auth.js [hunk 2]
```diff
@@ -27,18 +28,20 @@ export const mutations = {
...
+  login({ commit, dispatch }, payload) {
     authService.login(payload)
       .then(response => {
         const token = response.data.data
-        // console.log(response.data);
-        
+
         commit('AUTHENTICATE_USER', token)
 
         //det er her jeg skal sætte vores currentUser i vores user module til den user som er logget ind
```
Comment (Doommer43) [inline src/store/modules/auth.js]: Remove or change comment to english and use objective language.

File: src/components/Navigation.vue [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <template>
   <v-navigation-drawer v-model="drawer" fixed app>
-      <UserDisplay />
+      <UserDisplay :v-if='this.$store.state.user.currentUser.firstName == "admin" '    />
```
Comment (Doommer43) [inline src/components/Navigation.vue]: Is this assuming all administrative users will be named 'admin'?

File: src/components/UserDisplay.vue [hunk 4]
```diff
@@ -19,20 +19,36 @@ export default {
   name: 'UserDisplay',          
     data() {
         return{
-        user: this.$store.state.user.currentUser,
+        user: {}
```
Comment (Doommer43) [inline src/components/UserDisplay.vue]: Set dummy data, Firstname Lastname here, instead of empty object.

File: src/components/UserDisplay.vue [hunk 5]
```diff
@@ -19,20 +19,36 @@ export default {
...
-      }
-      else{
-        return this.user.firstName + " " +  this.user.lastName
-      }
+  methods: {     
+    
+    
+     //getting the name from the user and making sure there is something in it
+    getName() {
+          this.user = this.$store.state.user.currentUser
```
Comment (Doommer43) [inline src/components/UserDisplay.vue]: Make a 'created' function, retrieving and saving user data and change method accordingly.

File: src/components/UserDisplay.vue [hunk 6]
```diff
@@ -19,20 +19,36 @@ export default {
...
+            return "Placeholder Holdersen Place"
+          }
+          else{
+            return this.user.firstName + " " +  this.user.lastName
+       }
+    },
+
+    //getting picture for the avatar
+    getPicture() {
+      this.user = this.$store.state.user.currentUser
```
Comment (Doommer43) [inline src/components/UserDisplay.vue]: Do this in 'created' step of the lifecycle.

File: src/components/UserDisplay.vue [hunk 7]
```diff
@@ -19,20 +19,36 @@ export default {
...
-  methods: {
-    getName(){
-      console.log(this.user)
-      if(!this.user.firstName || !this.user.lastName || this.user.firstName === "" || this.user.lastName === ""){
-        return "Placeholder Holdersen Place"
-      }
-      else{
-        return this.user.firstName + " " +  this.user.lastName
-      }
+  methods: {
```
Comment (Doommer43) [inline src/components/UserDisplay.vue]: These methods should be made as computed values.

File: src/store/modules/user.js (hunk 1 above)
Comment (klementj) [reply src/store/modules/user.js]: Remember to look through all your changes before commit. That way you are less likely to get stuff like this
```suggestion
    state.currentUser =c urrentUser
```

File: src/store/modules/auth.js (hunk 2 above)
Comment (klementj) [reply src/store/modules/auth.js]: No longer need the comment. Please remove

File: src/components/Navigation.vue (hunk 3 above)
Comment (klementj) [reply src/components/Navigation.vue]: Good suggestion from doomer. We might have different user roles in the future.  

But is there really any need to check on, name, role, whatever to show the users name and picture. Ill suggest to remove the check

File: src/components/UserDisplay.vue (hunk 5 above)
Comment (klementj) [inline src/components/UserDisplay.vue]: Use descructuring

```suggestion
 const {firstname, lastname} = this.$store.state.user.currentUser
```

File: src/components/UserDisplay.vue (hunk 6 above)
Comment (klementj) [reply src/components/UserDisplay.vue]: Remove the logic when setting dummy-data in data() function

File: src/store/modules/auth.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import router from '@/router.js'
 import AuthService from '@/services/auth.service.js'
 import axios from 'axios'
+import { dispatch } from 'rxjs/internal/observable/pairs';
```
Comment (klementj) [inline src/store/modules/auth.js]: Please check your changes before commit and push. 

```suggestion
```

**Pull request merged**